src/Testcontainers.Pulsar/PulsarBuilder.cs

```diff
@@ -0,0 +1,95 @@
+
+namespace Testcontainers.Pulsar
+{
+  /// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+  [PublicAPI]
+  public sealed class PulsarBuilder : ContainerBuilder<PulsarBuilder, PulsarContainer, PulsarConfiguration>
+  {
+    public const string PulsarImage = "apachepulsar/pulsar-all:2.11.0";
+
+    public ushort PulsarPort = 6650;
+    public ushort PulsarAdminPort = 8080;
+    public ushort PulsarSQLPort = 8081;
+
+    public const string StartupScriptFilePath = "/testcontainers.sh";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="KafkaBuilder" /> class.
+    /// </summary>
+    public PulsarBuilder()
+        : this(new PulsarConfiguration())
+    {
+      DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="KafkaBuilder" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    private PulsarBuilder(PulsarConfiguration resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+      DockerResourceConfiguration = resourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override PulsarConfiguration DockerResourceConfiguration { get; }
+
+    /// <inheritdoc />
+    public override PulsarContainer Build()
+    {
+      this.Validate();
+      return new PulsarContainer(this.DockerResourceConfiguration, TestcontainersSettings.Logger);
+    }
+    /// <inheritdoc />
+    protected override PulsarBuilder Init()
+    {
+      return base.Init()
+          .WithImage(PulsarImage)
+          .WithPortBinding(this.PulsarPort, this.PulsarPort)
+          .WithPortBinding(this.PulsarAdminPort, this.PulsarAdminPort)
+          .WithPortBinding(this.PulsarSQLPort, this.PulsarSQLPort)
```

Comment:
```suggestion
          .WithPortBinding(PulsarPort, true)
          .WithPortBinding(PulsarAdminPort, true)
          .WithPortBinding(PulsarSQLPort, true)
```

We never bind static host ports, please use random assigned host ports instead. Your container can resolve the port with `GetMappedPublicPort(ushort)`, see:

- https://dotnet.testcontainers.org/api/best_practices/ (2. and 3.)
- https://dotnet.testcontainers.org/api/create_docker_container/#examples